### DISCUSSION TITLE
How Can I Tell If My Training Is Going Well Just by Checking Tensorboard ?

### DISCUSSION THREAD

**Discussion by lpscr** [p0p4k/vits2_pytorch#69] [Q&A] [answered]
hi  ,can you please explain this in a way that's easy to understand?

 i just make a lot test last days , this i don't understand

1    When should I stop the training process? or continue for  further steps for improvement?
2   How can I tell if my model is overfitting (learning too much from the training data)?
3   How can I recognize if my model isn't learning effectively, so I should stop the training to avoid wasting time?

I'm also unsure about when to stop, and I've spent many hours on this. Which curve or aspect of the training should I pay the most attention to?
i send some image if you you can mark how must be the good result be great 

![pcC](https://github.com/p0p4k/vits2_pytorch/assets/147736764/b256fb9e-7130-4183-92cc-8632c79daf32)

In the second picture, the elements should be close together for better understanding and time ?
![n2](https://github.com/p0p4k/vits2_pytorch/assets/147736764/ea3740c4-5c26-4644-ae9d-0f9dde985b94)

In the last picture, the yellow line should be a linear start 0.0 to 1.0 end ?
![n3](https://github.com/p0p4k/vits2_pytorch/assets/147736764/da91ff1e-49a5-43d4-bee8-ce906a008a0f)

If you could reply, I'd appreciate it.

Thank you so much

**Comment by p0p4k** [comment, marked as answer]
1 When should I stop the training process? or continue for further steps for improvement?
- check mel loss and kl loss. Mel loss is about reconstructing the wav given real mel data. KL loss is predicting the compressed mel representation given text. When they converge (become lines with no slope) you can stop and try to test.
2 How can I tell if my model is overfitting (learning too much from the training data)?
- Try to check using inference.py and use the checkpoints in logs to see how the samples sound.
3 How can I recognize if my model isn't learning effectively, so I should stop the training to avoid wasting time?
- Yes and probably try overfitting a single batch of data to see whether there is no bug in your training process and then fix them and try again.

**Reply by lpscr** [reply]
thank you very much for explaining :)

**Comment by p0p4k** [comment]
> In the last picture, the yellow line should be a linear start 0.0 to 1.0 end ?
- no, because the data is batched, the text and/or mel might not be the maximum size (first sample) of the batch.

**Reply by p0p4k** [reply]
but it should be monotonically increasing. which it always will be (since we use a pre-configured alignment algorithm with no gradients being passed through it)
